src/components/com_tjcertificate/administrator/models/certificates.php

```diff
@@ -81,18 +88,64 @@ protected function getListQuery()
 		// Initialize variables.
 		$db    = $this->getDbo();
 		$query = $db->getQuery(true);
-		$app = Factory::getApplication();
+		$app   = Factory::getApplication();
+		$user  = Factory::getUser();
 
 		$extension = Factory::getApplication()->input->get('extension', '', 'CMD');
 
 		$this->setState('filter.component', $extension);
 
+		// Filter by client
+		$client = $this->getState('filter.client');
+
 		// Create the base select statement.
 		$query->select(array('ci.*', 'ct.title', 'users.name as uname'));
 		$query->from($db->quoteName('#__tj_certificate_issue', 'ci'));
+
 		$query->join('LEFT', $db->quoteName('#__tj_certificate_templates', 'ct') .
 			' ON (' . $db->quoteName('ci.certificate_template_id') . ' = ' . $db->quoteName('ct.id') . ')');
-		$query->join('LEFT', $db->quoteName('#__users', 'users') . ' ON (' . $db->quoteName('ci.user_id') . ' = ' . $db->quoteName('users.id') . ')');
+
+		$query->join('INNER', $db->quoteName('#__users', 'users') .
+			' ON (' . $db->quoteName('ci.user_id') . ' = ' . $db->quoteName('users.id') . ')');
+
+		if (ComponentHelper::isEnabled('com_multiagency') && $this->params->get('enable_multiagency'))
+		{
+			$canManageAllAgencyUser = $user->authorise('core.manage.all.agency.user', 'com_multiagency');
+
+			// Subquery to get agency users
+			$subquery  = $db->getQuery(true);
+			$subquery->select($db->quoteName('ml.id'));
+			$subquery->from($db->quoteName('#__tjmultiagency_multiagency', 'ml'));
+			$subquery->join('INNER', $db->quoteName('#__tj_clusters', 'c') . ' ON ' . $db->quoteName('c.client_id') . '=' . $db->quoteName('ml.id'));
+			$subquery->join('INNER', $db->quoteName('#__tj_cluster_nodes', 'cn') . ' ON ' . $db->quoteName('cn.cluster_id') . '=' . $db->quoteName('c.id'));
+			$subquery->Where($db->qn('ml.state') . '=' . 1);
+			$subquery->where($db->quoteName('cn.user_id') . ' = ' . (int) $user->id);
+
+			$query->select('agency.title as title');
+
+			$query->join('INNER', $db->qn('#__tj_cluster_nodes', 'nodes') .
+				' ON (' . $db->qn('users.id') . ' = ' . $db->qn('nodes.user_id') . ')');
+
+			$query->join('INNER', $db->qn('#__tj_clusters', 'clusters') .
+				' ON (' . $db->qn('clusters.id') . ' = ' . $db->qn('nodes.cluster_id') .
+				' AND ' . $db->qn('clusters.client') . " = 'com_multiagency' )");
+
+			$query->join('LEFT', $db->qn('#__tjmultiagency_multiagency', 'agency') .
+				' ON (' . $db->qn('agency.id') . ' = ' . $db->qn('clusters.client_id') . ')');
+
+			$agencyId = $this->getState('filter.agency_id');
+
+			// If don't have manage all user permission then get users of own agency
+			if (!$canManageAllAgencyUser && !$agencyId)
+			{
```

Comment:
put all $subquery related statement inside this IF statement

Reply:
done